### DISCUSSION TITLE
JavaScript localization CDN addon on self-hosted Weblate

### DISCUSSION THREAD

**Discussion by PingvinB** [WeblateOrg/weblate#8248] [Q&A] [answered]
Hi, 

I'm trying to enable the [JavaScript localization CDN addon](https://docs.weblate.org/en/weblate-4.14.1/admin/addons.html#javascript-localization-cdn) but I can't find it the list of addons under Component -> Manage -> Addons. 
It's supposedly a built-in addon since 4.2, but does this apply to self-hosted Weblate as well? I'm am running Weblate 4.14.1.

I tried setting the [LOCALIZE_CND_URL and LOCALIZE_CND_PATH](https://docs.weblate.org/en/weblate-4.14.1/admin/config.html#localize-cdn-url-and-localize-cdn-path) environment variables thinking the missing vars are the reason for the addon not being in the list, but it didn't help. 

I also tried installing it manually with the install_addon command
`docker-compose exec --user weblate weblate-docker_weblate_1 weblate install_addon --addon weblate.cdn.cdnjs my-project/my-component --config '{"threshold": 0, "css_selector": "hmmwat"}'`
but doesn't look like anything happens - there's no feedback. 

Debug mode is on, but I don't see any error messages related to this.

**Comment by nijel** [comment]
The add-on is available only with the configured storage and URL (see https://docs.weblate.org/en/weblate-4.14.1/devel/html.html#weblate-cdn). Maybe you have typo in variable names? At least it's wrong in your post, it's `CDN` not `CND`.

Also, the add-on needs a monolingual component with at least one translation.

**Comment by PingvinB** [comment]
Hi! I just double-checked and they're only mistyped here in the github post. 
Maybe the values I set are wrong? Do you have a working example?

I do have the component set up with base language + 2 target languages.

**Reply by nijel** [reply]
The add-on does not really check the values. It will allow installation once anything is configured for both. The PATH is used for storing the CDN files, the URL is used for generating JavaScript snippets.

**Comment by PingvinB** [comment, marked as answer]
It seems I was supposed to prefix the environment variables with `WEBLATE_` in `docker-compose.override.yml` i.e. `WEBLATE_LOCALIZE_CDN_URL` and `WEBLATE_LOCALIZE_CDN_PATH`. 
Now the add-on shows up in the list. 

Thanks @nijel

**Reply by nijel** [reply]
Yes, the settings in Docker environment are prefixed with `WEBLATE_`
